### PR TITLE
Release v0.42.0

### PR DESCRIPTION
**Pull request by khanhtc1202**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

**Comment by khanhtc1202** [comment]
@knanao plz, when you have time 👀

File: docs/content/en/blog/releases/v0.42.0.md [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+description: >
+ Release v0.42.0
+---
+
+## Changes since v0.41.4
+
+### New Features
+
+* Add last use of api key on webui ([#4124](https://github.com/pipe-cd/pipecd/pull/4124))
+* Add pipectl deployment list ([#4182](https://github.com/pipe-cd/pipecd/pull/4182))
```
Comment (knanao) [inline docs/content/en/blog/releases/v0.42.0.md:14]: I think it's more clear to move to Notable changes as well as list applications by labels. wdyt?

Comment (khanhtc1202) [reply docs/content/en/blog/releases/v0.42.0.md]: for sure, updated 👌

**Review by knanao — APPROVED** [review]
Here you go!

File: docs/content/en/blog/releases/v0.42.0.md [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+* Add last use of api key on webui ([#4124](https://github.com/pipe-cd/pipecd/pull/4124))
+* Add diff details with terraform dritf detection ([#4167](https://github.com/pipe-cd/pipecd/pull/4167))
+
+### Notable Changes
+
+* Support installing pipectl via asdf ([#4185](https://github.com/pipe-cd/pipecd/pull/4185))
+* Add pipectl deployment list ([#4182](https://github.com/pipe-cd/pipecd/pull/4182))
+* Allow pipectl command to specify number of fetching applications ([#4189](https://github.com/pipe-cd/pipecd/pull/4189))
+* Support pipectl to list applications by Label. ([#4186](https://github.com/pipe-cd/pipecd/pull/4186))
```
Comment (khanhtc1202) [inline docs/content/en/blog/releases/v0.42.0.md:21]:
```suggestion
* Support pipectl to list applications by labels ([#4186](https://github.com/pipe-cd/pipecd/pull/4186))
```

**Comment by khanhtc1202** [comment]
@knanao sorry, could you please re-approve 🙏

**Review by knanao — APPROVED** [review]

**Pull request merged**